### ISSUE TITLE
Migrate timezone value from feedback sessions to courses

### ISSUE THREAD

**Issue by wkurniawan07** [TEAMMATES/teammates#5928]
This will still be on hold until #5863 is done, but surfacing the issue now since this will be subjected to some discussions. I have also been working on this, but obviously cannot create any PR first since the said holding PR needs to be merged first.

The starter menu will be:
1. Should we retain the timezone value in feedback sessions? Here are the pros/cons:
   - (+) Change to code is not too much since we have been relying on timezone value in sessions. Without it, we have to get the timezone value from the course object and this may result in extra query (although querying a course object should not be expensive).
   - (+) We can tune the timezone up to the session level (not sure if this is an important feature to have or not).
   - (-) We can tune the timezone up to the session level, meaning whatever setting in the course level will be neglected (and thus, what's the point?)
2. After migration, how should we inform the instructors of the new behaviour? This is not as easy as "your course is configured at UTC+8 timezone" because UTC+8 corresponds to tens of cities in the world. We're going to have something like "UTC+8 -> Asia/Singapore" mapping, and thus we're bound to have instructors with courses configured at wrong timezones (unless we have the timezone settings for each, but I don't believe we have).

**Comment by damithc** [comment]
See if this has any relevant info . https://dl.dropboxusercontent.com/u/4223019/Ashreay_Fyp_Report.pdf

**Comment by wkurniawan07** [comment]
Unfortunately no, whatever is written there has been done in #5863. No info on how to handle the migration and its post-mortem.

**Comment by damithc** [comment]
In terms of display, we can show both `[UTC+8] Asia Singapore`.
We can continue to show the time zone in the session page, although the value will come from the course object.

**Comment by wkurniawan07** [comment]
Display is the least of my concern. The problem comes when we use the timezone ID to determine offset, and especially so when we start handling DST.
Because we can only do a one-to-one mapping from an offset to a timezone ID, there is no way we can do an accurate migration. Let's say all "UTC+8" are migrated to "Asia/Singapore", then courses in e.g "Australia/Perth" (also in UTC+8) may not have 100% accuracy when they are observing DST.

**Comment by damithc** [comment]
Um, we should not migrate existing sessions. The two systems will co-exist for a while. Only the courses created in future will have the new behavior. Does that address your concern?

**Comment by damithc** [comment]
@wkurniawan07 any thoughts on how to proceed with this?

**Comment by whipermr5** [comment]
> After migration, how should we inform the instructors of the new behaviour? This is not as easy as "your course is configured at UTC+8 timezone" because UTC+8 corresponds to tens of cities in the world. We're going to have something like "UTC+8 -> Asia/Singapore" mapping, and thus we're bound to have instructors with courses configured at wrong timezones (unless we have the timezone settings for each, but I don't believe we have).

We can map to the Etc/GMT+8 time zone, which is a legitimate time zone ID not affected by DST.

**Comment by wkurniawan07** [comment]
@whipermr5 that's the path of least surprise. May not be the best but I guess there's not much we can do.

**Comment by whipermr5** [comment]
Some discussion in another PR beginning here: https://github.com/TEAMMATES/teammates/pull/8750#issuecomment-379003464
